### ISSUE TITLE
Error when running Numpy, Scipy, Pandas code with Windows installer

### ISSUE THREAD

**Issue by minatouma** [spyder-ide/spyder#20242]
### What steps will reproduce the problem?

I'm running pybamm using DFN model but I get this error No module named 'numpy.core._multiarray_umath, I have updated python version to 10.8. Why I'm getting this issue? any resolution ? thanks 

## Versions

* Spyder version: 5.3.3  (conda)
* Python version: 3.10.8 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.7
* Operating System: Windows 10

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.31.1;<8.0.0       :  7.31.1 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.1 (OK)
jellyfish >=0.7               :  0.9.0 (OK)
jsonschema >=3.2.0            :  4.16.0 (OK)
keyring >=17.0.0              :  23.4.0 (OK)
nbconvert >=4.0               :  6.5.4 (OK)
numpydoc >=0.6.0              :  1.5.0 (OK)
paramiko >=2.4.0              :  2.8.1 (OK)
parso >=0.7.0;<0.9.0          :  0.8.3 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.9.0 (OK)
pygments >=2.0                :  2.11.2 (OK)
pylint >=2.5.0;<3.0           :  2.14.5 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.5.0;<1.6.0          :  1.5.0 (OK)
pylsp_black >=1.2.0           :  1.2.1 (OK)
qdarkstyle >=3.0.2;<3.1.0     :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.1.10 (OK)
qtawesome >=1.0.2             :  1.0.3 (OK)
qtconsole >=5.3.2;<5.4.0      :  5.3.2 (OK)
qtpy >=2.1.0                  :  2.2.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  65.5.0 (OK)
sphinx >=0.6.6                :  5.0.2 (OK)
spyder_kernels >=2.3.3;<2.4.0 :  2.3.3 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.6 (OK)
zmq >=22.1.0                  :  23.2.0 (OK)

# Optional:
cython >=0.21                 :  None (NOK)
matplotlib >=3.0.0            :  None (NOK)
numpy >=1.7                   :  None (NOK)
pandas >=1.1.1                :  None (NOK)
scipy >=0.17.0                :  None (NOK)
sympy >=0.7.3                 :  1.11.1 (OK)
```

**Comment by dalthviz** [comment]
Hi @minatouma seems like you don't have `numpy` installed in the env Spyder is runing. Could you try to install and check again?

**Comment by emmanuel-contreras** [comment]
I'm running into this error trying to follow [the Ray getting started tutorial](https://docs.ray.io/en/latest/ray-overview/getting-started.html) I am wondering if it's an issue with spyder or my environment or something else. Happy to start a new issue if it's not related to this one. Using the standalone spyder installer and a conda env called _ml_

Screenshot:
* top left: spyder trying to run the first few lines of the tutorial
* bottom left: ray dashboard.err file, first few lines pointing to spyder numpy
* right side: same lines of code running fine outside spyder in the same conda environment.

Please let me know if you need any more info!

This is the line that throws the error trying to load the ray dashboard
``` Python
# Load data.
dataset = ray.data.read_csv("s3://anonymous@air-example-data/breast_cancer.csv")
```

![image](https://user-images.githubusercontent.com/14666496/222467909-4d1fac30-e00d-49dc-ab0e-0a33d7db6d82.png)


https://docs.ray.io/en/latest/ray-overview/getting-started.html

--- 
Python 3.8.10 64-bit 
Qt 5.15.2 
PyQt5 5.15.7 
Windows 10 

# Mandatory:
atomicwrites >=1.2.0                 :  1.4.1 (OK)
chardet >=2.0.0                      :  5.1.0 (OK)
cloudpickle >=0.5.0                  :  2.2.0 (OK)
cookiecutter >=1.6.0                 :  2.1.1 (OK)
diff_match_patch >=20181111          :  20200713 (OK)
intervaltree                         :  None (OK)
IPython >=7.31.1;<9.0.0              :  8.8.0 (OK)
jedi >=0.17.2;<0.19.0                :  0.18.2 (OK)
jellyfish >=0.7                      :  0.9.0 (OK)
jsonschema >=3.2.0                   :  4.17.3 (OK)
keyring >=17.0.0                     :  23.13.1 (OK)
nbconvert >=4.0                      :  7.2.8 (OK)
numpydoc >=0.6.0                     :  1.5.0 (OK)
paramiko >=2.4.0                     :  2.12.0 (OK)
parso >=0.7.0;<0.9.0                 :  0.8.3 (OK)
pexpect >=4.4.0                      :  4.8.0 (OK)
pickleshare >=0.4                    :  0.7.5 (OK)
psutil >=5.3                         :  5.9.4 (OK)
pygments >=2.0                       :  2.14.0 (OK)
pylint >=2.5.0;<3.0                  :  2.15.10 (OK)
pylint_venv >=2.1.1                  :  2.3.0 (OK)
pyls_spyder >=0.4.0                  :  0.4.0 (OK)
pylsp >=1.7.1;<1.8.0                 :  1.7.1 (OK)
pylsp_black >=1.2.0                  :  1.2.1 (OK)
qdarkstyle >=3.0.2;<3.1.0            :  3.0.3 (OK)
qstylizer >=0.2.2                    :  0.2.2 (OK)
qtawesome >=1.2.1                    :  1.2.2 (OK)
qtconsole >=5.4.0;<5.5.0             :  5.4.0 (OK)
qtpy >=2.1.0                         :  2.3.0 (OK)
rtree >=0.9.7                        :  1.0.1 (OK)
setuptools >=49.6.0                  :  66.0.0 (OK)
sphinx >=0.6.6                       :  6.1.3 (OK)
spyder_kernels >=2.4.2;<2.5.0        :  2.4.2 (OK)
textdistance >=4.2.0                 :  4.5.0 (OK)
three_merge >=0.1.1                  :  0.1.1 (OK)
watchdog                             :  2.2.1 (OK)
zmq >=22.1.0                         :  24.0.1 (OK)

# Optional:
cython >=0.21                        :  0.29.33 (OK)
matplotlib >=3.0.0                   :  3.6.3 (OK)
numpy >=1.7                          :  1.24.1 (OK)
pandas >=1.1.1                       :  1.5.2 (OK)
scipy >=0.17.0                       :  1.10.0 (OK)
sympy >=0.7.3                        :  1.11.1 (OK)

# Spyder plugins:
spyder_terminal.terminalplugin 1.2.2 :  1.2.2 (OK)

**Comment by dalthviz** [comment]
Hi @minatouma thank you for the feedback! What happens if you import `numpy` from the Spyder IPython console and check its version and location? Running something like:

```python
import numpy

numpy.__version__
num.__file__

```

The info shown from the package corresponds with your `ml` env?

**Comment by emmanuel-contreras** [comment]
Hi sorry I'm not @minatouma but thought my issue may be related.

When I import numpy and check the version I get the expected values in the conda _ml_ environment and in the right location. But you can see below in spyder console that I get the umath error if I try to load the ray dataset that starts the dashboard

![image](https://user-images.githubusercontent.com/14666496/222530612-85635c53-0e11-43e5-85c3-7108441ba871.png)

**Comment by emmanuel-contreras** [comment]
Here a screenshot, I've never modified this screen so everything should be the default.  
![image](https://user-images.githubusercontent.com/14666496/222743766-0345f92e-af4e-4af9-81f3-4e105fc4bef3.png)

**Comment by dalthviz** [comment]
Oh I see, I think the `ray` code to read the dataset should work if you uncheck the `pkgs` directory there @emmanuel-contreras could you check?

I think what is happening here is that the standalone installer adds to the system `PYTHONPATH` the `pkgs` dir but then we need to check the order in how the `PYTHONPATH manager` is adding the `System PYTHONPATH` in that case or maybe we need to add some logic to automatically deactivate that path if using a custom interpreter inside the standalone installer to prevent accidentally shadowding modules inside the env from the `pkgs` directory (and then causing errors like the one you are experiencing).

Pinging @ccordoba12 maybe he has some ideas about how to fix this.

**Comment by emmanuel-contreras** [comment]
that worked! yea that makes sense, so is there any harm in leaving that PYTHONPATH unchecked so that only the modules in my env get used? I always point spyder to a different conda interpreter when I use it

![image](https://user-images.githubusercontent.com/14666496/222829643-76af00fc-42cc-466a-a0cb-222ed8c9b6ff.png)

**Comment by dalthviz** [comment]
Awesome! 🎉 As long as you are using a custom Python interpreter it should be fine to leave the `pkgs` dir unchecked I think :+1:

**Comment by ccordoba12** [comment]
> or maybe we need to add some logic to automatically deactivate that path if using a custom interpreter inside the standalone installer to prevent accidentally shadowding modules inside the env from the pkgs directory

Yeah, that's what I also think we need to do, i.e. automatically disable the `pkgs` directory from our Pythonpath manager when (i) we detect Spyder was installed through our Windows installer and (ii) the user sets a custom interpreter. That way we'll avoid this problem completely.

**Comment by LuYupeng311** [comment]
> Yeah, that's what I also think we need to do, i.e. automatically disable the `pkgs` directory from our Pythonpath manager when (i) we detect Spyder was installed through our Windows installer and (ii) the user sets a custom interpreter. That way we'll avoid this problem completely.

Does this mean that the packages in the 'pkgs' directory in the Windows installer can no longer be imported and used normally? But I don't want to install these packages again in the custom interpreter environment, which will take up storage space repeatedly and be troublesome.

**Comment by ccordoba12** [comment]
> Does this mean that the packages in the 'pkgs' directory in the Windows installer can no longer be imported and used normally?

It means that if you're using an external environment, you'll need to install Numpy, Pandas, Scipy, Matplotlib, etc, in that environment for those packages to work. That has always been the case, it's just that a refactoring we did introduce this error.

But, if you're using the default interpreter, then you'll still be able to work with those packages without problems.

**Comment by mrclary** [comment]
@emmanuel-contreras, What version of Spyder are you using?

@ccordoba12 @dalthviz, This issue may already be resolved by #20106, which fixed several issues with getting the system PYTHONPATH (after 5.4.2 release). Can you confirm with the latest pull request artifact (https://github.com/spyder-ide/spyder/pull/20656/checks)?

**Comment by dalthviz** [comment]
I think you are right @mrclary ! Checked installing the latest artifact generated and over the `PYTHONPATH manager` I'm not seeing the standalone `pkgs` dir and I'm able to use a custom interpreter with the `ray` code :+1:

**Comment by emmanuel-contreras** [comment]
@mrclary I am using 5.4.2, so it's been fixed on the upcoming version? I'll be sure to update when it's officially out. Thank you!

**Comment by ccordoba12** [comment]
I'm going to close this issue as solved by #20106, according to what @dalthviz said above.
